### PR TITLE
Adds owners and applications ACL in Worf command line and interactive

### PR DESCRIPTION
**Pull request by slambour**

### CODE REVIEW

File: warp10/src/main/java/io/warp10/worf/EncodeTokenCommand.java [hunk 1]
```diff
@@ -59,7 +66,19 @@ public boolean execute(String function, WorfKeyMaster worfKeyMaster, PrintWriter
         out.println("token=" + token);
         out.println("tokenIdent=" + worfKeyMaster.getTokenIdent(token));
         out.println("application name=" + application);
-        out.println("producer & owner=" + producer + " & " + owner);
+        out.println("producer=" + producer);
+
+        switch (tokenType) {
+          case READ:
+            out.println("authorizations: applications" + applications);
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/worf/EncodeTokenCommand.java]: No space between the text and the list?

File: warp10/src/main/java/io/warp10/worf/WorfCLI.java [hunk 2]
```diff
@@ -159,13 +174,64 @@ public int execute(String[] args) {
...
+
+          // test the validity of the owner uuid
           UUID.fromString(ownerUID);
+
         } else {
           ownerUID = producerUID;
         }
 
+        // owners empty ? add current owner to the authorized owners ()
+        if (authorizedOwnersUID== null || authorizedOwnersUID.isEmpty()) {
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/worf/WorfCLI.java]: null == authorizedOwnersUID

File: warp10/src/main/java/io/warp10/worf/WorfCLI.java [hunk 3]
```diff
@@ -159,13 +174,64 @@ public int execute(String[] args) {
...
+          // test if the appname have a pattern x{y}
+          Matcher matcher = tokenPattern.matcher(appName);
+
+          // extract app names of pattern app{app,app1,app2}
+          if (matcher.matches()) {
+            authorizedApplications = new ArrayList<>();
+            String[] apps = matcher.group(2).split(",");
+
+            // adds uuid to the list, fail otherwise
+            for (String app:apps){
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/worf/WorfCLI.java]: Add whitespace 'app: apps) {'

File: warp10/src/main/java/io/warp10/worf/WorfCLI.java [hunk 4]
```diff
@@ -159,13 +174,64 @@ public int execute(String[] args) {
...
+              authorizedApplications.add(URLDecoder.decode(app, "UTF-8"));
+            }
+
+            appName = URLDecoder.decode(matcher.group(1), "UTF-8");
+          }
+
+          if (authorizedApplications==null || authorizedApplications.isEmpty()) {
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/worf/WorfCLI.java]: null == authorizedApplications

File: warp10/src/main/java/io/warp10/worf/WorfCLI.java [hunk 5]
```diff
@@ -179,6 +245,27 @@ public int execute(String[] args) {
...
+            case "w":
+              tokenType = WorfTokenType.WRITE;
+              break;
+
+            case "rw":
+              tokenType = WorfTokenType.READ_WRITE;
+              break;
+
+            default:
+              throw new WorfException("Invalid token type : " + tt);
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/worf/WorfCLI.java]: Get rid of whitespace -> 'type: '

File: warp10/src/main/java/io/warp10/worf/WorfInteractive.java [hunk 6]
```diff
@@ -510,6 +556,42 @@ private String getUUID(String line, PrintWriter out, String worfDefault, String
...
+      // Check uuid integrity
+      //
+      ArrayList<String> output = new ArrayList<String>();
+      for (String strUuid : uuids) {
+        UUID.fromString(strUuid);
+        output.add(strUuid);
+      }
+      return output;
+    } catch( Exception exp) {
+      if (out!=null) {
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/worf/WorfInteractive.java]: null != out

File: warp10/src/main/java/io/warp10/worf/WorfInteractive.java [hunk 7]
```diff
@@ -510,6 +556,42 @@ private String getUUID(String line, PrintWriter out, String worfDefault, String
...
+      //
+      // Check uuid integrity
+      //
+      ArrayList<String> output = new ArrayList<String>();
+      for (String strUuid : uuids) {
+        UUID.fromString(strUuid);
+        output.add(strUuid);
+      }
+      return output;
+    } catch( Exception exp) {
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/worf/WorfInteractive.java]: Fix whitespace

File: warp10/src/main/java/io/warp10/worf/WorfInteractive.java [hunk 8]
```diff
@@ -25,6 +25,7 @@
 import jline.console.completer.StringsCompleter;
 import org.apache.commons.lang.math.NumberUtils;
 import org.apache.commons.math3.util.Pair;
+import org.boon.core.Sys;
```
Comment (hbs) [inline warp10/src/main/java/io/warp10/worf/WorfInteractive.java]: Where is boon used?

File: warp10/src/main/java/io/warp10/worf/EncodeTokenCommand.java (hunk 1 above)
Comment (slambour) [reply warp10/src/main/java/io/warp10/worf/EncodeTokenCommand.java]: Updated with = separator (output homogeneity)

File: warp10/src/main/java/io/warp10/worf/WorfInteractive.java (hunk 8 above)
Comment (slambour) [reply warp10/src/main/java/io/warp10/worf/WorfInteractive.java]: boon unused IDE auto import / line removed

**Pull request merged**